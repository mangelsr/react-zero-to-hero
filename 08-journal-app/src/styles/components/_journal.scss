.journal__main-content {
    display: flex;
}

.journal__sidebar {
    background-color: $dark-gray;
    color: white;
    display: flex;
    flex-direction: column;
    height: 100vh;
    padding: 0 10px;
    width: 450px;
}

.journal__sidebar-navbar {
    display: flex;
    justify-content: space-between;
}

.journal__sidebar-navbar h3 {
    font-weight: lighter;
}

.journal__new-entry {
    align-items: center;
    cursor: pointer;
    display: flex;
    justify-content: center;
    margin-top: 30px;
    width: 100%;
    flex-direction: column;

    transition: color .3s ease;

    &:hover {
        color: darken($color: white, $amount: 20);
    }
}

.journal__entries {
    flex: 1 1 auto;
    margin-top: 10px;
    overflow-y: auto;
}

.journal__entry {
    background-color: white;
    border-radius: 4px;
    color: $dark-gray;
    display: flex;
    margin-bottom: 10px;
    overflow: hidden;
    justify-content: space-between;
}

.journal__entry-picture {
    height: 75px;
    width: 75px;

}

.journall__entry-body {
    padding: 10px;
}

.journall__entry-title {
    font-size: 14px;
    font-weight: bold;
}

.journall__entry-content {
    font-size: 10px;
}

.journall__entry-date-box {
    align-items: center;
    display: flex;
    flex-direction: column;
    justify-content: center;
    padding: 5px;
}

.journall__entry-date-box span {
    font-size: 12px;
}