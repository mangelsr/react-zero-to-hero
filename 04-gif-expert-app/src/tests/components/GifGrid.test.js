import '@testing-library/jest-dom';
import React from 'react'
import { shallow } from 'enzyme';

import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('Pruebas sobre <GifGrid />', () => {
   
    test('Debe mostrarse el componente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifGrid category='Shaman King' />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar items al cargar imagenes', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier.gif',
            title: 'Titulo'
        }];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category='Shaman King' />);
        
        // expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    });
    
});