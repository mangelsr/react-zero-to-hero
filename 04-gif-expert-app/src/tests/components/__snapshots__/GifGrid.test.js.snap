// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <GifGrid /> Debe mostrarse el componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Shaman King
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
